### PR TITLE
Get specific performance counters.

### PR DESCRIPTION
**Pull request by mateoatr**

### CODE REVIEW

File: AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs [hunk 1]
```diff
@@ -21,14 +21,27 @@ public class PerformanceCollectorModule : ITelemetryModule
...
 
             foreach (string counter in defaultCounters)
                 factory.GetCounter(counter);
         }
+
+        /// <summary>
+        /// Get specific performance counters that are not in the default counters list.
+        /// </summary>
+        /// <param name="counterName"> Counter name.</param>
+        public void AddCounter(params string[] counterName)
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs]: you already have counters collection ptroperty. Use it

File: AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs [hunk 2]
```diff
@@ -21,14 +24,40 @@ public class PerformanceCollectorModule : ITelemetryModule
                                                                 @"\Processor(_Total)\% Processor Time"
                                                             };
 
-        public IList<PerformanceCounterCollectionRequest> Counters { get; private set; }
+        public List<PerformanceCounterCollectionRequest> Counters { get; private set; }
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs]: Why did you change it?

File: AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs [hunk 3]
```diff
@@ -21,14 +24,40 @@ public class PerformanceCollectorModule : ITelemetryModule
...
 
             foreach (string counter in defaultCounters)
-                factory.GetCounter(counter);
+            {
+                try
+                {
+                    ICounterValue c = factory.GetCounter(counter);
+                    configuration.RegisterCounter(c);
+                }
+                catch (ArgumentException e)
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs]: this is strange to catch and rethrow exception without any other code or comment

Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs]: You probably meant to trace it instead of re-throwing

File: AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs (hunk 2 above)
Comment (mateoatr) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs]: @SergeyKanzhelev I was getting this error: 

> Element 'Counters' element does not have a Type attribute, does not specify a value and is not a valid collection type. Type initialization was skipped.

Comment (dnduffy) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs]: This kind of return type needs to be interface and not concrete, furthermore, List is specifically supposed to be for internal use and not exposed directly (there is an FxCop rule for this).

File: AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs [hunk 4]
```diff
@@ -21,14 +24,40 @@ public class PerformanceCollectorModule : ITelemetryModule
...
+                {
+                    throw e;
```
Comment (dnduffy) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs]: When you catch and then re-throw the exception directly you lose the stack trace up to that point and start over with the new throw.  If, however, you do just "throw;" and don't specify the exception explicitly the stack trace is preserved.  In this case there is nothing happening so the whole catch is pointless.

File: AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs [hunk 5]
```diff
@@ -21,14 +24,40 @@ public class PerformanceCollectorModule : ITelemetryModule
...
+            foreach (var counter in Counters)
+            {
+                try
+                {
+                    ICounterValue c = factory.GetCounter(counter.PerformanceCounter);
+                    configuration.RegisterCounter(c);
+                }
+                catch (ArgumentException e)
+                {
+                    throw e;
```
Comment (dnduffy) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs]: Same here, don't catch and throw for no purpose.  It is okay to catch, add value, and then re-throw.  "Add value" could be logging and "throw;" to preserve the stack trace or adding additional information and throwing a new wrapping exception.

File: AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs (hunk 2 above)
Comment (dnduffy) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs]: @mateoatr the error could be because there is no default initialization of the list variable in a constructor.  Making collections as properties directly settable is generally a bad design.
